tests/test_dtypes.py

```diff
@@ -0,0 +1,157 @@
+    check(
+        assert_type(pd.IntervalDtype(np.timedelta64), pd.IntervalDtype),
+        pd.IntervalDtype,
+    )
+    check(
+        assert_type(pd.IntervalDtype(np.datetime64), pd.IntervalDtype), pd.IntervalDtype
+    )
+
+
+def test_int64_dtype() -> None:
+    check(assert_type(pd.Int8Dtype(), pd.Int8Dtype), pd.Int8Dtype)
+    check(assert_type(pd.Int16Dtype(), pd.Int16Dtype), pd.Int16Dtype)
+    check(assert_type(pd.Int32Dtype(), pd.Int32Dtype), pd.Int32Dtype)
+    check(assert_type(pd.Int64Dtype(), pd.Int64Dtype), pd.Int64Dtype)
+    check(assert_type(pd.UInt8Dtype(), pd.UInt8Dtype), pd.UInt8Dtype)
+    check(assert_type(pd.UInt16Dtype(), pd.UInt16Dtype), pd.UInt16Dtype)
+    check(assert_type(pd.UInt32Dtype(), pd.UInt32Dtype), pd.UInt32Dtype)
+    check(assert_type(pd.UInt64Dtype(), pd.UInt64Dtype), pd.UInt64Dtype)
+
+    i64dt = pd.Int64Dtype()
+    check(assert_type(i64dt.itemsize, int), int)
+    check(assert_type(i64dt.na_value, NAType), NAType)
+    check(assert_type(i64dt.is_signed_integer, bool), bool)
+    check(assert_type(i64dt.is_unsigned_integer, bool), bool)
+    check(assert_type(i64dt.numpy_dtype, np.dtype), np.dtype)
+    check(assert_type(i64dt.construct_array_type(), "type[IntegerArray]"), type)
+
+
+def test_categorical_dtype() -> None:
+    cdt = pd.CategoricalDtype(categories=["a", "b", "c"], ordered=True)
+    check(assert_type(cdt, pd.CategoricalDtype), pd.CategoricalDtype)
+    check(
+        assert_type(pd.CategoricalDtype(categories=[1, 2, 3]), pd.CategoricalDtype),
+        pd.CategoricalDtype,
+    )
+    check(assert_type(cdt.categories, pd.Index), pd.Index)
+    assert check(assert_type(cdt.ordered, Optional[bool]), bool)
+
+
+def test_sparse_dtype() -> None:
+    s_dt = pd.SparseDtype("i4")
+    check(assert_type(s_dt, pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(str), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(complex), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(bool), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(np.int64), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(str), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(float), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(np.datetime64), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(np.timedelta64), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype("datetime64"), pd.SparseDtype), pd.SparseDtype)
+    check(assert_type(pd.SparseDtype(), pd.SparseDtype), pd.SparseDtype)
+    # pyright ignore because mypy does not like non-minimal unions, while pyright
+    # can't minimize to check
+    check(
+        assert_type(
+            s_dt.fill_value,  # pyright: ignore[reportGeneralTypeIssues]
+            Union[str, bytes, dt.date, timedelta, complex, None],
```

Comment:
why not just use `Union[Scalar, None]` here.

Reply:
Changed.